src/osx/cocoa/dataview.mm

```diff
@@ -2158,12 +2158,21 @@ -(BOOL) becomeFirstResponder
 {
     bool handled = wxWidgetCocoaImpl::doCommandBySelector(sel, slf, _cmd);
     // if this special key has not been handled
-    if ( !handled && IsInNativeKeyDown() )
+    if ( !handled )
     {
-        // send the original key event back to the native implementation to get proper default handling like eg for arrow keys
-        wxOSX_EventHandlerPtr superimpl = (wxOSX_EventHandlerPtr) [[slf superclass] instanceMethodForSelector:@selector(keyDown:)];
-        superimpl(slf, @selector(keyDown:), GetLastNativeKeyDownEvent());
+        if ( sel == @selector(cancelOperation:) )
+        {
+            [m_OutlineView cancelOperation: slf];
+        }
+
+        if ( IsInNativeKeyDown() )
+        {
+            // send the original key event back to the native implementation to get proper default handling like eg for arrow keys
+            wxOSX_EventHandlerPtr superimpl = (wxOSX_EventHandlerPtr) [[slf superclass] instanceMethodForSelector:@selector(keyDown:)];
+            superimpl(slf, @selector(keyDown:), GetLastNativeKeyDownEvent());
+        }
```

Comment:
Sure, I can commit this instead. I'd like to add a check for `superimpl != nullptr` before using it however as I'm not sure at all that it can't be null.